### ISSUE TITLE
typing_extensions.Literal isn't treated the same as typing.Literal

### ISSUE THREAD

**Issue by sminnee** [microsoft/pyright#3936]
**Describe the bug**
Using typing_extensions.Literal as a type signature should do the same thing as typing.Literal. Instead, typing_extensions.Literal is treated as a normal class.

**To Reproduce**

The `django-types` package imports Literal from typing_extensions, which I believe is necessary for older versions of python.

Example: https://github.com/sbdchd/django-types/blob/main/django-stubs/db/models/fields/related.pyi#L47

This is used with `@overloads` matching Literal[True] and Literal[False] on a `null` init param in the same field. 

https://github.com/sbdchd/django-types/blob/main/django-stubs/db/models/fields/related.pyi#L116
https://github.com/sbdchd/django-types/blob/main/django-stubs/db/models/fields/related.pyi#L154

However, when I set `null=True`, rather than matching against the right overload, I get this error:

```py
contact_activity = ForeignKey(
        WorkOrderContactActivity, on_delete=models.DO_NOTHING, related_name="+", null=True, blank=True
    )
```

```
Argument of type "Literal[True]" cannot be assigned to parameter "null" of type "Literal" in function "__init__"
  "Literal[True]" is incompatible with "Literal"Pylance[reportGeneralTypeIssues](https://github.com/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md#diagnostic-severity-rules)
```
It appears as though it's treating typing_extensions.Literal as a regular class, as opposed to a literal. Changing to typing.Literal fixes this, but I believe that this would break compatibility with older python versions.

**Expected behavior**

Using either of these as type signatures should have the same effect.

**Comment by sminnee** [comment]
Here's a minimal replication case

```py
from typing_extensions import Literal

def my_func(arg: Literal[True]):
    return arg

my_func(True)
```

**Comment by sminnee** [comment]
typing_extensions.py has the following:

```py
# 3.8+:
if hasattr(typing, 'Literal'):
    Literal = typing.Literal
# 3.7:
else:
    class _LiteralForm(typing._SpecialForm, _root=True):

        def __repr__(self):
            return 'typing_extensions.' + self._name

        def __getitem__(self, parameters):
            return typing._GenericAlias(self, parameters)

    Literal = _LiteralForm('Literal',
                           doc="""A type that can be used to indicate to type checkers
                           that the corresponding value has a value literally equivalent
                           to the provided parameter. For example:

                               var: Literal[4] = 4

                           The type checker understands that 'var' is literally equal to
                           the value 4 and no other value.

                           Literal[...] cannot be subclassed. There is no runtime
                           checking verifying that the parameter is actually a value
                           instead of a type.""")
```

**Comment by erictraut** [comment]
I'm not able to repro the problem you're seeing. I suspect that you have an old version of the `typing_extensions` stub library installed in your environment. Please uninstall that if it's present, and your issue should go away.

**Comment by sminnee** [comment]
Ah, it turns out that setting the stubs path to `/usr/local/lib/python3.9/site-packages` in your VSCode PyLance config broke things for me.
